chore: Update noir

sirasistant opened the pull request.

TomAFrench commented on noir/compiler/noirc_evaluator/src/ssa/opt/bubble_up_constrains.rs:
This isn't necessary as any duplicate constraints are going to be handled by https://github.com/noir-lang/noir/pull/4060

TomAFrench commented on noir/compiler/noirc_evaluator/src/ssa/opt/bubble_up_constrains.rs:
Just saw PR in the main repo. I was about to ask that we avoid making changes in fix PRs so thanks!

sirasistant replied:
Basically bubble up is reversing the order for multiple assertions on the same data. Without it, tests fail because it uses the last assertion instead of the first, here you can see the errors: https://github.com/AztecProtocol/aztec-packages/pull/4136

sirasistant replied:
Yes I added it here to see if it fixes all cases!

TomAFrench reviewed (approved):
LGTM

sirasistant commented:
Oops, forgot to update the aztec-packages branch on noir. Doing that now!

sirasistant committed:
git subrepo pull (merge) noir

subrepo:
  subdir:   "noir"
  merged:   "602f23f4f"
upstream:
  origin:   "https://github.com/noir-lang/noir"
  branch:   "aztec-packages"
  commit:   "602f23f4f"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

sirasistant committed:
chore: update yarn lock
